File: burger-maker-service/src/main/java/com/zenika/tutoriels/observabilitydemo/burgermaker/clients/CheeseServiceClient.java

package com.zenika.tutoriels.observabilitydemo.burgermaker.clients;

import com.zenika.tutoriels.observabilitydemo.api.contracts.CheeseService;
import org.springframework.cloud.openfeign.FeignClient;

@FeignClient(value = "${services.fridge.name:fridge}", contextId = "cheese",
        url = "${services.fridge.url:}", configuration = FridgeErrorClientConfiguration.class)
public interface CheeseServiceClient extends CheeseService {
}

File: burger-maker-service/src/main/kotlin/com/zenika/tutoriels/observabilitydemo/burgermaker/clients/CheeseServiceClient.kt
package com.zenika.tutoriels.observabilitydemo.burgermaker.clients

import org.springframework.web.reactive.function.client.WebClient
import reactor.core.publisher.Mono

class CheeseServiceClient(serviceUrl: String, builder: WebClient.Builder) : AbstractFridgeClient(builder, "${serviceUrl}/cheeses") {
    fun orderCheeseByKind(kind: String): Mono<Void> {
        return delete(kind)
    }
}